obspy/signal/konnoohmachismoothing.py

```diff
@@ -210,11 +228,8 @@ def konno_ohmachi_smoothing(spectra, frequencies, bandwidth=40, count=1,
     # spectrum is to be smoothed.
     if enforce_no_matrix is False and (len(spectra.shape) > 1 or count > 1) \
        and approx_mem_usage < max_memory_usage:
-        # Disable NumPy warnings due to possible divisions by zero/logarithms
-        # of zero.
-        with np.errstate(all='ignore'):
-            smoothing_matrix = calculate_smoothing_matrix(
-                frequencies, bandwidth, normalize=normalize)
+        smoothing_matrix = calculate_smoothing_matrix(
+            frequencies, bandwidth, normalize=normalize)
         new_spec = np.dot(spectra, smoothing_matrix)
         # Eventually apply more than once.
         for _i in range(count - 1):
```

Comment:
Unless you want to avoid that spectra dtype check you could use the new function here, to avoid code duplication and give it a bit of logical structure..